Sources/DualCameraKit/VideoRecording/DualCameraReplayKitVideoRecorder.swift

```diff
@@ -40,9 +40,23 @@ public actor DualCameraReplayKitVideoRecorder: DualCameraVideoRecording {
         
         let outputURL = configure(outputURL: config.outputURL)
         
-        recorder.startRecording()
-        state = .active(outputURL: outputURL)
-        DualCameraLogger.session.debug("📹 Screen recording started with ReplayKit")
+        try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Void, Error>) in
+            recorder.startRecording { error in
+                if let error = error {
+
+                    if let nsError = error as? NSError, nsError.domain == RPRecordingErrorDomain {
```

Comment:
test this out on device more. it seems to be flaky whether the `DualCameraError.permissionDenied` localize desc shows